### ISSUE TITLE
Komodo load wrong module when debugging two applications

### ISSUE THREAD

**Issue by jrheisey** [Komodo/KomodoEdit#1294]
### Short Summary
The secondary project Windows is loading the version of the module for the primary project when debugging.

### Steps to Reproduce
* Need two versions of the same project.
  * Besides a primary script file each project has a separate module.pm which is used.
  * The modules are the same name in each project.
* In my case the file names are the same just the paths are different. One path is on a different drive letter with the folder structure similar with a difference in one extra folder node or depth.
  * i.e. c:\projects\tools\mytool and u:\projects\release\tools\mytool
* Start Komodo and open a project
* Click File => New => New Window
* Open the alternate project located in the alternate path
* Set break points on the same statement in each project.
  * In my case the identical statements are on different line numbers.
* Run the primary project and have it halt at the break point.
* Run the secondary project 
  * In my case it opens the version of the file from the primary target and the primary targets breakpoint triggers from the secondary project window.
* I have altered each verion by adding 'printf("mymodule.pm 1\n") at the top. Of course the secondary project has a 2 in the string.
* I tried using the Perl -I argument in the secondary project Debugging Options to have Perl reference the correct mymodule.pm folder. It did not work.
* Launching from the secondary project windows always uses the wrong module file.

### Expected results
It seems it happens during or after the primary project window is in a debug session.
Use the correct module file.

### Actual results
See above

### Platform Information
*Komodo Edit or IDE?* IDE
*Komodo Version?* Komodo IDE, version 9.3.2, build 88191, platform win32-x86.
Built on Thu Dec 10 11:38:17 2015.
*Operating System (and version)?* Windows 7 64 bit

### Additional Information
I thought about renaming on of the modules but the module contains a couple of packages using the module name. I would be impractical to rename all the references to the package name.

**Comment by Naatan** [comment]
Could you rephrase this part? I'm not entirely sure I understand what's happening:

> In my case it opens the version of the file from the primary target and the primary targets breakpoint triggers from the secondary project window.

Thanks!

**Comment by jrheisey** [comment]
The 'version of the file' I am referring to the mymodule.pm file. Specifically the version marked with the printf("mymodule.pm 1\n") statement.

When I run the secondary application from secondary Windows it does run the correct application Perl script but the wrong mymydule.pm 1 is also loaded. Once the breakpoint is triggered both the primary and secondary Windows track when single stepping from the secondary window.

**Comment by jrheisey** [comment]
I took some screen shots at different points during the sequence.
![komodo 1 primary start dialog](https://cloud.githubusercontent.com/assets/4934544/14763413/ee65072e-0949-11e6-9119-5422a41bd64a.png)
* Start first application simt4.pl using dwarf.pm
![komodo 2 secondary start dialog](https://cloud.githubusercontent.com/assets/4934544/14763414/f17a3826-0949-11e6-961c-652f01de91fa.png)
* Start second altered application simt4.pl using dwarf.pm
![komodo 3 secondary break trigger using wrong module](https://cloud.githubusercontent.com/assets/4934544/14763416/f3d12ea4-0949-11e6-8cda-9438bf3ca02a.png)
* Secondary application open the dwarf.pm from the primary's folder.
![komodo 4 after single step from secondary window](https://cloud.githubusercontent.com/assets/4934544/14763417/f60152d0-0949-11e6-9f65-333a9a50864b.png)
* After a single step from the secondary application you can see that both windows have updated.

**Comment by Naatan** [comment]
Are you sure the Directory field is different in both debugging sessions?

**Comment by jrheisey** [comment]
Yes with a caveat. I see an issue when I try to display the Debugging Options dialog for both sessions at the same time by selecting 'Go' in each session.
* Select 'Debug -> Go/Continue' from primary window to display the 'Debugging Options' dialog 
* Select 'Debug -> Go/Continue' from secondary window
  * This results in the primary session's dialog being populate with the debugging field data from secondary window's project debugging data.
  * I would expect a second dialog to appear with the secondary project's debugging field data.

**Comment by Naatan** [comment]
Is that a new issue you're reporting or are you saying this is what happens as a leadup to the issue you're reporting?

**Comment by jrheisey** [comment]
Initially the script primary script was running before starting the secondary script.
It is a variation of the previous set of steps which exposes another perhaps related anomaly.
You could consider this a separate bug. If it is related this could be a clue to help you determine a root cause.

**Comment by jrheisey** [comment]
OK I have something else going on with my startup parameters that may not be related to having multiple projects open while debugging.
I have been editing my Dwarf.pm module then starting the debugger from this editor context.
These startup parameters appear to be stored somewhere else other than the current project.

When I open and run my secondary project as the only opened project I see the wrong Dwarf.pm file being opened. Where are these debugging options stored?

**Comment by mitchell-as** [comment]
It sounds like there's a Perl path problem if the "wrong" Dwarf.pm file is being loaded from your secondary project (without the first project being loaded).

As far as I know, the only debugging options are stored in the Go/Run dialog that you posted in your screenshots.

**Comment by jrheisey** [comment]
I figured it out.
I had previously set a project specific path for Perl paths in File -> Preferences -> Languages -> Perl -> Additional Perl Import Directories.
I moved the folder name into the project specific configuration.
Sorry for the false alarm.

However there still is the issue when trying display two 'Debuggging Options' dialogs at the same time.
One for the primary window and one for the secondary window.

**Comment by Naatan** [comment]
@jrheisey thanks for the update! 

Could you open a new bug report for the remaining issue? Just to avoid any confusion.

**Comment by jrheisey** [comment]
You got it.
FYI: I got a clue when I looked in Task manager and enabled the 'Command Line' column on the 'Processes' tab. I could see the -I parameters pointing to the wrong folder.

**Comment by Naatan** [comment]
Thanks @jrheisey - much appreciated! :)

**Comment by tonyattwood** [comment]
Just  take a look at  this stuff, it's reaaly good, more info here <http://fair.heavyliftingsoftware.com/e4spddha>


My best to you, tony.attwood@ntlworld.com

**Comment by tonyattwood** [comment]
Hey,

I've recently visited a nice  place,  just take  a look, you're gonna love  it for sure! Here are some pics of  it <http://fact.travelingbills.com/e4owervk>

Sent from a prehistoric stone tablet, tony.attwood@ntlworld.com
